### DISCUSSION TITLE
Having issues with createSignal's setter typing

### DISCUSSION THREAD

**Discussion by sveinnthorarins** [solidjs/solid#725] [Q&A]
Hey!

I'm creating a context and want to pass in a default value to get rid of the possibility of an `undefined` return type from `useContext`.

The context's value/store is just a boolean signal (i.e. the returned getter and setter duo from a `createSignal<boolean>`).
```ts
type EditModeStore = [Accessor<boolean>, Setter<boolean>];
```
I'm trying to create the context like this (supplying some basic functions as the default value):
```ts
const EditModeContext = createContext<EditModeStore>([() => false, (b) => false]);
```
When I do this, TypeScript gives me the following error about my default setter's return value:
```
Type 'boolean' is not assignable to type 'U'.
  'boolean' is assignable to the constraint of type 'U', but 'U' could be
  instantiated with a different subtype of constraint 'boolean'.
```
Note, the `Setter<boolean>` typing looks like this:
```ts
<U extends boolean>(v: (U extends Function ? never : U) | ((prev: boolean) => U)) => U
```
Can anyone help me out here?

How can I supply a default function that conforms to the `Setter<boolean>` typing?

**Comment by milomg** [comment]
The issue is that the type Setter specifies that you will also support a function instead of a value. Here's an example of a function that works: 
```ts
const EditModeContext = createContext<EditModeStore>([
  () => false,
  (b) => (typeof b == "function" ? b() : b),
]);
```

**Reply by sveinnthorarins** [reply]
Sorry for the late reply. Thanks for your help! 
You were correct but I also had to supply an argument to the function call `b()`.
So this is what I ended up with (it works):
```ts
const EditModeContext = createContext<EditModeStore>([
    () => false, 
    (b) => (typeof b === 'function' ? b(false) : b)
]);
```
